tests/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -242,7 +243,12 @@ public function testListTablesWithFilter(string $prefix, int $expectedCount): vo
         $this->markConnectionNotReusable();
 
         $this->connection->getConfiguration()->setSchemaAssetsFilter(
-            static function (string $name) use ($prefix): bool {
+            /** @param AbstractAsset|string $name */
+            static function ($name) use ($prefix): bool {
+                if (! is_string($name)) {
```

Comment:
It doesn't look like a documentation change. Why does this method need to be capable of filtering objects? So far it's been capable of filtering names, which seems to be fine.

Reply:
The documentation change helps static analysis spot a flaw in this method, which may indeed be call with a `Sequence` object: https://github.com/doctrine/dbal/blob/d72299248a9d4c52b05a3d3815a116d063b5528d/src/Schema/AbstractSchemaManager.php#LL193C86-L193C86

Reply:
I still don't understand why we need to add the change in the logic as part of the documentation change. I also don't understand why the filtering function should allow all non-strings in the collection. What's the point of passing an `AbstractAsset[]` to this implementation?

Reply:
That test case is named `testListTablesWithFilter`, In the case of tables, only strings are passed when calling `filterAssetNames()`: https://github.com/doctrine/dbal/blob/19f0dec95edd6a3c3c5ff1d188ea94c6b7fc903f/src/Schema/AbstractSchemaManager.php#L335-L360

So this implementation seems not to care at all about sequences. In that case, I chose not to filter them out then, but I might as well have thrown an exception since it never happens in the test.

> What's the point of passing an AbstractAsset[] to this implementation?

You mean in the code snippet in https://github.com/doctrine/dbal/pull/6071#discussion_r1237339714 ?

I don't know, I didn't write it… maybe the point is allowing users to use `AbstractAsset::getNamespaceName()` for filtering?